std/stdio.d

```diff
@@ -466,6 +466,283 @@ Throws: `ErrnoException` if the file could not be opened.
+         */
+        existingOnly = 1 << 5,
+    }
+
+    private static bool hasAnyWriteFlag(OpenMode openMode) @safe nothrow pure @nogc
+    {
+        return (openMode & (OpenMode.write | OpenMode.truncate | OpenMode.append | OpenMode.createNew)) != 0;
+    }
+
+    private static const(char)[4] modezFromOpenMode(OpenMode openMode) @trusted
+    {
+        const bool anyWrite = hasAnyWriteFlag(openMode);
+        char[4] modez;
+        size_t i = 0;
+        if (openMode & OpenMode.read) {
+            if (anyWrite) {
+                if (openMode & OpenMode.append) {
+                    modez[i++] = 'a';
+                    modez[i++] = '+';
+                } else if (openMode & OpenMode.existingOnly) {
+                    modez[i++] = 'r';
+                    modez[i++] = '+';
+                } else if (openMode & OpenMode.truncate) {
+                    modez[i++] = 'r';
+                    modez[i++] = 'w';
+                } else {
+                    modez[i++] = 'w';
+                    modez[i++] = '+';
+                }
+            } else {
+                modez[i++] = 'r';
+            }
+        } else if (anyWrite) {
+            if (openMode & OpenMode.append)
+                modez[i++] = 'a';
+            else
+                modez[i++] = 'w';
+        }
+        assert(i < modez.length);
+        modez[i] = '\0';
+        return modez;
+    }
+
+    version(Posix) private static int openFd(string name, OpenMode openMode) @trusted
+    {
+        import std.exception : errnoEnforce;
+        import std.internal.cstring : tempCString;
+        import std.conv : octal;
+        import core.sys.posix.fcntl : O_RDWR, O_RDONLY, O_WRONLY, O_TRUNC, O_APPEND, O_EXCL, O_CREAT, mode_t;
+        static import core.sys.posix.fcntl;
+
+        mode_t mode;
+        const bool anyWrite = hasAnyWriteFlag(openMode);
+        const bool existing = (openMode & OpenMode.existingOnly) != 0;
+        const bool hasRead = (openMode & OpenMode.read) != 0;
+        if (hasRead && anyWrite)
+            mode |= O_RDWR;
+        else if (hasRead)
+            mode |= O_RDONLY;
+        else if (anyWrite)
+            mode |= O_WRONLY;
+
+        if (anyWrite && !existing)
+            mode |= O_CREAT;
+
+        if (openMode & OpenMode.truncate)
+            mode |= O_TRUNC;
+        if (openMode & OpenMode.append)
+            mode |= O_APPEND;
+        if (openMode & OpenMode.createNew)
+            mode |= O_EXCL;
+
+        auto namez = name.tempCString();
+        int fd;
+        if (mode & O_CREAT)
+            fd = core.sys.posix.fcntl.open(namez, mode, octal!666);
+        else
+            fd = core.sys.posix.fcntl.open(namez, mode);
+        errnoEnforce(fd >= 0);
+        return fd;
+    }
+
+    version(Windows) private static HANDLE openHandle(string name, OpenMode openMode) @trusted
+    {
+        import std.internal.cstring : tempCString;
+        import core.sys.windows.windows;
```

Comment:
selective import, as you did for the Posix version of the function

Reply:
It's too many symbols. But will do if it's really required.

Reply:
I think you don't have the choice if you want tthis PR to become green Circle CI (still) fails because of this.